### PR TITLE
[connectors] Implement Zendesk connector deletion

### PR DESCRIPTION
**Pull request by aubin-tchoi**
## Description

- Implement Zendesk connector + related resources deletion (`clean` method on the `ConnectorManager`).
This change allows deleting it from poke.

## Deploy Plan

- Deploy connectors.

### CODE REVIEW

File: connectors/src/connectors/zendesk/index.ts [hunk 1]
```diff
@@ -99,7 +99,21 @@ export class ZendeskConnectorManager extends BaseConnectorManager<null> {
...
   async clean(): Promise<Result<undefined, Error>> {
-    throw new Error("Method not implemented.");
+    const { connectorId } = this;
+    const connector = await ConnectorResource.fetchById(connectorId);
+    if (!connector) {
+      logger.error({ connectorId }, "[Zendesk] Connector not found.");
+      return new Err(new Error("Connector not found"));
+    }
+
+    const result = await connector.delete();
```
Comment (Fraggle) [inline connectors/src/connectors/zendesk/index.ts:109]: Does that automatically delete all related datasources ?

Comment (Fraggle) [reply connectors/src/connectors/zendesk/index.ts:109]: After looking at the code of google drive, it seems that you need to implements more in `ZendeskConnectorStrategy` (maybe it's planned in another PR?)

Comment (aubin-tchoi) [reply connectors/src/connectors/zendesk/index.ts:109]: No, this is handled by front; when it calls the endpoint `connectors/delete/:connector_id` it also calls `coreAPI.deleteDataSource`

Comment (aubin-tchoi) [reply connectors/src/connectors/zendesk/index.ts:109]: yes this is planned in another PR 👍

Comment (aubin-tchoi) [reply connectors/src/connectors/zendesk/index.ts:109]: done! (implemented the deletion of the resources associated to a connector in `ZendeskConnectorStrategy.delete`).

**Review by Fraggle — APPROVED** [review]
LGTM.

**Comment by Fraggle** [comment]
>No, AFAIC this is handled by front; when it calls the endpoint connectors/delete/:connector_id (which runs the method clean added here) it also calls coreAPI.deleteDataSource

Note, unrelated to your work, i'm surprised that connector isn't in charge to do it. (as the connector is in charge to add it)

**Pull request merged**